### DISCUSSION TITLE
Is Temperature Behavior During 0.5C Charging in SPMe Model Normal?

### DISCUSSION THREAD

**Discussion by ErfanSamadi1998** [pybamm-team/PyBaMM#4648] [Q&A] [answered]
I have been working with the following code:

options = {"thermal": "lumped"}
model = pybamm.lithium_ion.SPMe(options=options)
parameter_values = pybamm.ParameterValues("OKane2022")

I observed that when charging the battery at 0.5C, the cell temperature initially increases, then decreases slightly around the middle of the charging process, and finally increases again, as shown in the attached plot.

Is this behavior typical for batteries, or could it be related to inaccuracies or limitations in the model?
![Screenshot 2024-12-08 072505](https://github.com/user-attachments/assets/c98bbb5e-2747-4779-ada6-be365aaea57c)

**Comment by ALMA7867** [comment]
mine is also mostly same. however it changes with cell volume and cooling surface area for lumped thermal model.
![image](https://github.com/user-attachments/assets/ca449b56-2b38-4361-b738-3e904995499f)

**Comment by rtimms** [comment, marked as answer]
Yes, this can be a correct scenario depending on your cooling conditions. Irreversible heating is typically larger at high and low SOC, so it is reasonable that you may see a temperature decrease in the middle where the thermal boundary condition wins out over the internal heat generation.

Try playing with the script below 

```python3
import pybamm
import matplotlib.pyplot as plt

options = {"thermal": "lumped"}
model = pybamm.lithium_ion.SPMe(options=options)
parameter_values = pybamm.ParameterValues("OKane2022")
parameter_values["Total heat transfer coefficient [W.m-2.K-1]"] = pybamm.InputParameter(
    "h"
)
experiment = pybamm.Experiment(["Discharge at 0.5C until 2.8V", "Rest for 1 hour"])
solver = pybamm.IDAKLUSolver()
sim = pybamm.Simulation(
    model, parameter_values=parameter_values, experiment=experiment, solver=solver
)
sols = {}
fig, ax = plt.subplots(1, 1)
for h in [0, 10, 50]:
    sol = sim.solve(inputs={"h": h})
    sols[h] = sol
    ax.plot(
        sol["Time [s]"].data,
        sol["Volume-averaged cell temperature [K]"].data,
        label=f"h = {h}",
    )
ax.legend()
ax.set_xlabel("Time [s]")
ax.set_ylabel("Temperature [K]")
ax.set_title("Temperature")

for h, sol in sols.items():
    fig, ax = pybamm.plot_thermal_components(sol, show_plot=False)
    fig.suptitle(f"h = {h} W/m^2/K")
    fig.tight_layout()
plt.show()
```

**Reply by ErfanSamadi1998** [reply]
Thank you very much for the explanation! Now I completely understand the reason behind this behavior. 🙏🙏
